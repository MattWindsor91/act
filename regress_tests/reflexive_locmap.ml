(* The Automagic Compiler Tormentor

   Copyright (c) 2018--2019 Matt Windsor and contributors

   ACT itself is licensed under the MIT License. See the LICENSE file in the
   project root for more information.

   ACT is based in part on code from the Herdtools7 project
   (https://github.com/herd/herdtools7) : see the LICENSE.herd file in the
   project root for more information. *)

(** Tests the output of the 'act-c gen-locmap' command for generating
    reflexive diff location maps. *)

open Core
module Ac = Act_common

let gen_locmap_for_file ~(file : Fpath.t) ~(path : Fpath.t) : unit Or_error.t =
  ignore file ;
  Or_error.(
    (* TODO(@MattWindsor91): this duplicates a lot of the command it's
       supposed to be testing. *)
    path
    |> Plumbing.Input.of_fpath
    |> Act_c_mini.Frontend.load_from_isrc
    >>| Act_c_mini.Litmus.vars
    >>| Act_backend.Diff.Location_map.reflexive
    >>= Act_backend.Diff.Location_map.output ~onto:Plumbing.Output.stdout
  )

let run (test_dir : Fpath.t) : unit Or_error.t =
  let full_dir = Fpath.(test_dir / "litmus" / "") in
  Common.regress_on_files "Reflexive-locmap" ~dir:full_dir ~ext:"litmus"
    ~f:gen_locmap_for_file

let command =
  Common.make_regress_command ~summary:"runs delitmusifier regressions" run

let () = Command.run command
