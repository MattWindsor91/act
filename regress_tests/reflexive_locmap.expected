# Reflexive-locmap tests

## iriw.c.litmus

```
{
  "x": "x",
  "y": "y",
  "2:r0": "2:r0",
  "2:r1": "2:r1",
  "3:r0": "3:r0",
  "3:r1": "3:r1"
}
```
## iriw_fuzz_redundant.c.litmus

```
{
  "MaJgQmON3yu": "MaJgQmON3yu",
  "VFTQNWN": "VFTQNWN",
  "ZgoV7uel9o": "ZgoV7uel9o",
  "m02M5s": "m02M5s",
  "ne0J8wrPQam": "ne0J8wrPQam",
  "x": "x",
  "y": "y",
  "3:r0": "3:r0",
  "3:r1": "3:r1",
  "4:r0": "4:r0",
  "4:r1": "4:r1"
}
```
## memalloy_cmpxchg.c.litmus

```
{ "x": "x", "0:r0": "0:r0" }
```
## memalloy_na_int.c.litmus

```
{ "x": "x", "0:r0": "0:r0" }
```
## rwc.c.litmus

```
{ "x": "x", "y": "y", "1:r0": "1:r0", "1:r1": "1:r1", "2:r0": "2:r0" }
```

Ran 5 test(s).
